### PR TITLE
Clarify quickstart editions

### PR DESCRIPTION
**Pull request by dadlerj**
Test plan:

### CODE REVIEW

File: doc/index.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Sourcegraph documentation
 
-[Sourcegraph](https://sourcegraph.com) is a web-based, open-source, self-hosted code search and navigation tool for developers, used by Uber, Lyft, Yelp, and more.
+[Sourcegraph](https://sourcegraph.com) is a web-based, open-core, self-hosted code search and navigation tool for developers, used by Uber, Lyft, Yelp, and more.
```
Comment (ryan-blunden) [inline doc/index.md]: What does `open-core` mean?

Comment (ryan-blunden) [inline doc/index.md]: I had never come across the the term **open-core** before so I would suggest linking to an authoritative source. 

Is it really necessary to change it from **open-source**?

Comment (ryan-blunden) [inline doc/index.md]:
```suggestion
[Sourcegraph](https://about.sourcegraph.com) is a web-based, open-core, self-hosted code search and navigation tool for developers, used by Uber, Lyft, Yelp, and more.
```

We should link to the about site as someone logged-in to Sourcegraph.com won't be redirected.

File: doc/index.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 
 ## Quickstart guide
 
-It takes less than 5 minutes to run and install Sourcegraph using Docker:
+It takes less than 5 minutes to run and install [Sourcegraph Core](https://about.sourcegraph.com/pricing) for free using Docker:
```
Comment (ryan-blunden) [inline doc/index.md]:
```suggestion
It takes less than 5 minutes to deploy [Sourcegraph Core](https://about.sourcegraph.com/pricing) for free using Docker:
```

File: doc/index.md (hunk 1 above)
Comment (dadlerj) [reply doc/index.md]: Yeah, re-reading this just sounds confusing. I'll revert the change for now. This statement is a bit too sweeping, but with the clarifications below, I'm okay with this as a temporary state..

Comment (dadlerj) [reply doc/index.md]: Actually, removing it entirely for now

File: doc/index.md (hunk 2 above)
Comment (dadlerj) [reply doc/index.md]: I see what your comment here is about, but I actually like "run and install" as "deploy" seems more like jargon we use internally, it sounds like a heavyweight process, and this quickstart guide actually **IS** for them to run it locally (not to "deploy" to a remote VM or anything, which is what that word evokes in my mind...).

LMK if you disagree.

**Pull request merged**